File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) 
    {
        int temp=0,i,j,ans=0;
        for(i=0,j=height.length-1;i<height.length-1&&j>i;)
        {   
            if(height[i]<height[j])
            {
                temp=(j-i)*height[i];
                i++;
            }
            else
            {
                temp=(j-i)*height[j];
                j--;
            }
            if(temp>=ans)
            ans=temp;
        }
        return ans;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int back=h.size()-1;
        int front=0;
        int max=0,maximum=0;
        while(back>front)
        {
            if(h[back]>h[front])
            {
            max=h[front]*(back-front);
            }
            else
            {
            max=h[back]*(back-front);
            }
            if(max>maximum)
            {
                maximum=max;
            }
            if(h[front]<h[back])
                front++;
            else
                back--;

            
            
        }
        return maximum;
        
    }
};